docs/src/index.md

```diff
@@ -27,7 +27,271 @@ this library has a few goals:
+
+```scala
+import cats.parse.Rfc5234.{sp, alpha}
+import cats.parse.Parser
+
+/* Product */
+
+// the sp parser won't return the whitespace, it just returns Unit if it successful
+val p1: Parser[(Char, Unit)] = alpha ~ sp
+
+p1.parse("t")
+// res0: Either[Error, Tuple2[String, Tuple2[Char, Unit]]] = Left(Error(1,NonEmptyList(InRange(1, , ))))
+p1.parse("t ")
+// res1: Either[Error, Tuple2[String, Tuple2[Char, Unit]]] = Right((,(t,())))
+
+/* productL, productR */
+
+// The type is just Char because we dropping the space
+// to drop the alphabet change the arrow side: alpha *> sp
+val p2: Parser[Char] = alpha <* sp
+
+// still error since we need the space even if we drop it
+p2.parse("t")
+// res2: Either[Error, Tuple2[String, Char]] = Left(Error(1,NonEmptyList(InRange(1, , ))))
+p2.parse("t ")
+// res3: Either[Error, Tuple2[String, Char]] = Right((,t))
+
+/* OrElse */
+
+val p3: Parser[AnyVal] = alpha | sp
+
+p3.parse("t")
+// res4: Either[Error, Tuple2[String, AnyVal]] = Right((,t))
+p3.parse(" ")
+// res5: Either[Error, Tuple2[String, AnyVal]] = Right((,()))
+```
+
+## Repeating parsers
+
+Sometimes we need something to repeat zero or more types. The cats-parse have `rep` and `rep0` methods for repeating values. `rep` means that the parser must be successful *at least one time*. `rep0` means that the parser output might be empty.
+
+```scala
+import cats.data.NonEmptyList
+import cats.parse.Rfc5234.alpha
+import cats.parse.Parser
+
+val p1: Parser[NonEmptyList[Char]]  = alpha.rep
+val p2: Parser0[List[Char]] = alpha.rep0
+
+p1.parse("")
+// Left(Error(0,NonEmptyList(InRange(0,A,Z), InRange(0,a,z))))
+p2.parse("")
+// Right((,List()))
+p2.parse("something")
+// Right((,List(s, o, m, e, t, h, i, n, g)))
+```
+
+Notice the types of parsers. `Parser` type always means some non-empty output and the output of `Parser0` might be empty.
+
+## Parsers with empty output
+
+Some parsers never return a value. They have a type `Parser0`. One might get this type of parser when using `rep0` or `.?` methods.
+
+```scala
+import cats.parse.Rfc5234.{alpha, sp}
+import cats.parse.Parser
+
+val p: Parser[String] = (alpha.rep <* sp.?).rep.map { words => 
+  words.map(charsList => charsList.toList.mkString("")).toList.mkString(" ")
+}
+
+p.parse("hello world")
+// res0 = Right((,hello world))
+```
+
+Notice the type we got - `Parser[String]`. That is because we have `rep` outside and our `alpha.rep` parser with `Parser` type is on the left side of the clause. But what if we want to parse strings with spaces at the beginning?
+
+```scala
+val p = (sp.? *> alpha.rep <* sp.?).rep.map { words => 
+  words.map(charsList => charsList.toList.mkString("")).toList.mkString(" ")
+}
+```
+
+We will get an error `value rep is not a member of cats.parse.Parser0`. This happens since we have the left-side parser as optional in `sp.? *> alpha.rep <* sp.?` clause. This clause has a type `Parser0` which can't be repeated.
```

Comment:
maybe we should not that we can do `alpha.repAs[String]` or `alpha.rep.string` (in this particular case, although not in general, they are the same). That is better than `rep` + `map` because we don't make the intermediate list in that case. We actually use a `StringBuilder` internally to build up the result.